neo/Consensus/ConsensusContext.cs

```diff
@@ -279,7 +349,24 @@ public void Reset(byte viewNumber)
                     NextConsensus = Blockchain.GetConsensusAddress(NativeContract.NEO.GetValidators(Snapshot).ToArray()),
                     ConsensusData = new ConsensusData()
                 };
+                var pv = Validators;
                 Validators = NativeContract.NEO.GetNextBlockValidators(Snapshot);
+                if (_witnessSize == 0 || (pv != null && pv.Length != Validators.Length))

```

Comment:
I did not understand this part of the witness limit very much.
I will double check in 1-2 days with @igormcoelho.

Reply:
Is a recalculation, if the Validators change, we need to recompute the expected witness size

Reply:
3 days @vncoelho , any news?

Reply:
All right.